_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,332 @@
+- Jest documentation links to the RTL website in the
+[*Testing Web Framework*](https://jestjs.io/docs/testing-frameworks#react) section as well as React documentation in the
+ [*Recommended Tools*](https://reactjs.org/docs/testing.html#tools).
+- Testing—library allows us to test with the same API not only React but also others frameworks like Vue
+ (testing—library/vue), Angular (tesiting—library/angular), Svelte (testing—library/svelte) etc. This could be some
+  advantage for people working on several projects in different technologies. There is also *dom-testing-library* which
+   allows writing tests without any framework!
+
+## Basic differences
+
+The main differences between the two libraries that I can notice:
+1. **Different approaches to testing**
+
+   What distinguishes RTL is that we find elements by their labels or texts, almost like a user would. The point is to
+    replicate and simulate the user’s behavior. RTL tries to render components to the DOM and doesn’t give us access
+     directly to the internal component. While Enzyme uses shallow rendering (or sometimes deep rendering) and
+      encourages us to test the components instance using its state and props. Enzyme also allows full rendering of DOM
+       but it is not the basic assumption of the library’s operation.
+2. **Age**
+
+   Enzyme is 6 years old, RTL is 3 years old
+
+## Comparison of downloads
+
+Below we can see a comparison of the number of downloads of both libraries in the last 5 years. Between August and
+ September 2020, the number of RTL downloads exceeded the number of Enzyme downloads.
+
+![Enzyme and RTL dowloands graph](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/06.png)
+*Source [npm trends](https://www.npmtrends.com/@testing-library/react-vs-enzyme)*
+
+In the case of RTL, it is clear that the number of downloads is still growing strongly. On the other hand, it can be
+ suspected that the highest peak of Enzyme downloads is already behind.
+
+## Reasons to change the testing library
+
+Well, let’s be honest, Enzyme.js is a powerful library which has helped us test our React Components in an easier way
+ for a long time. But maybe... too long? The first official release was in December 2015. Do not get me wrong, I don’t
+  want to say that everything old is bad. But the frontend world is changing constantly, just as the approach to
+   testing it. I know that the Enzyme.js library hasn’t been deprecated yet, but there are a lot of reasons to switch
+    your test library right now. **If you haven’t read it I highly recommend you to take a look at this
+     [Piotr Staniów article](https://www.piotrstaniow.pl/goodbye-enzyme)** (it takes about 10 minutes to read).
+
+The main reasons for me to think about slowly rewriting your tests and switch to another library from this article are:
+- one developer maintaining the entire package,
```

Comment:
```suggestion
- one developer maintaining the entire Enzyme package,
```
Akapit wyżej jest o Enzymie, ale by to jeszcze podkreśliła, że switch _z Enzyme_ na coś innego.